### PR TITLE
Added SecurityProfile to support Host Encryption for VM and Scaleset resources

### PR DESCRIPTION
**Pull request by hari-bodicherla**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by hari-bodicherla** [comment]
@majastrz , Could you review and approve the PR from ARM side as we need sign off from ARM since we are adding a new definition here. Thanks

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json [hunk 1]
```diff
@@ -7773,6 +7779,15 @@
       },
       "description": "Specifies the storage settings for the virtual machine disks."
     },
+    "SecurityProfile": {
+      "properties": {
+        "encryptionAtHost": {
+          "type": "boolean",
+          "description": "This property can be used by user in the request to enable or disable Host Encryption for the virtual machine or virtual machine scale set. This will enable the encryption for all the disks including Temp disk at host itself. <br><br> Default: The Encryption at host will be disabled unless this property is not set to true for the resource."
```
Comment (hyonholee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json:7786]: Isn't the opposite true?   The Encryption at host will be disabled unless this property "is set to true" for the resource.

**Comment by majastrz** [comment]
> @majastrz , Could you review and approve the PR from ARM side as we need sign off from ARM since we are adding a new definition here. Thanks

The current on-call (@chiragg4u) can review from ARM side. (It's a weekly rotation.)

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json [hunk 2]
```diff
@@ -7773,6 +7773,15 @@
       },
       "description": "Specifies the storage settings for the virtual machine disks."
     },
+    "SecurityProfile": {
+        "properties": {
+            "encryptionAtHost": {
```
Comment (chiragg4u) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json]: >encryptionAtHost [](start = 13, length = 16)

Looks like this is a new writable property being added. I need following info
1. Will this break old SDK's for this API version? Do you've telemetry/data to prove?
2. Can older SDK cause overwrite of this parameter value while doing GET/PUT?

**Review by chiragg4u — CHANGES_REQUESTED** [review]
:clock1:

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json (hunk 2 above)
Comment (hari-bodicherla) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json]: Hi Chirag,
1.  This change will not break for any old SDKs as this is a nullable property and SDKs using the the api version wont send this property.  Also once the swagger changes are merged, we will be generating the SDK changes with this property .
2. The older SDKs as mentioned above will be sending us this as null which is handled in our code and GET also will set this property as null . So there wont be any breaking changes.

Comment (chiragg4u) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json]: Ok the scenario is 
1. Someone from portal created VM with encryptionAtHost settings configured.
2. OLD SDK based code would GET and will not recieve this field.
3. OLD SDK based code would PUT and nullify this field. Corrrect?
This is data loss and would change the behavior of the object.

Comment (hari-bodicherla) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json]: Hi Chirag, 
In this Scenario, when the customer used old SDK for the PUT operation, the property will be null and our code will ignore this null property and dont update any settings for the encryptionathost. So the user will still retain the encryptionathost settings after the PUT request with old SDK.

**Review by chiragg4u — APPROVED** [review]

**Comment by hyonholee** [comment]
Resolve conflict and make it sure all CI tests pass (except for breaking change CI test)

**Comment by hari-bodicherla** [comment]
@hyonholee , I have resolved the conflicts, Can you please review and approve the PR. I will ping other reviewers too . Thanks

**Comment by MS-syh2qs** [comment]
@hari-bodicherla Please resolve the conflicts

File: specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json [hunk 3]
```diff
@@ -0,0 +1,11068 @@
...
+          "type": "string",
+          "description": "The disk name."
+        },
+        "vhd": {
+          "$ref": "#/definitions/VirtualHardDisk",
+          "description": "The virtual hard disk."
+        },
+        "image": {
+          "$ref": "#/definitions/VirtualHardDisk",
+          "description": "The source user image virtual hard disk. The virtual hard disk will be copied before being attached to the virtual machine. If SourceImage is provid
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json:7835]: pls. update Temp disk to Resource/Temp disk.

Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-01/compute.json:7838]: virtual machine or virtual machine scale set.

**Comment by Sandido** [comment]
Looks like the number of Files Changed is now 0. Looks like your previous changes were lost. Please update this PR with all your changes.

**Pull request merged**